### PR TITLE
fixing doxygen

### PR DESCRIPTION
**Pull request by huguesBouvier**
fixing doxygen for POSIX and fixing bug in POSIX cond

Description
-----------
fixing doxygen for POSIX and fixing bug in POSIX cond

Checklist:
----------

- [x ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: doc/config/posix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+HTML_OUTPUT = POSIX
+
+# Generate Doxygen tag file for this library.
+GENERATE_TAGFILE = doc/tag/POSIX.tag
+
+# Directories containing library source code.
+INPUT = doc \
+        doc/lib \
+        libraries/freertos_plus/standard/freertos_plus_posix/include \
+        libraries/freertos_plus/standard/freertos_plus_posix/source \
```
Comment (yuhui-zheng) [inline doc/config/posix]: FreeRTOS_POSIX_utils.c under this dir is better not to be documented.

File: doc/config/posix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Library documentation output directory.
+HTML_OUTPUT = POSIX
+
+# Generate Doxygen tag file for this library.
+GENERATE_TAGFILE = doc/tag/POSIX.tag
+
+# Directories containing library source code.
+INPUT = doc \
+        doc/lib \
+        libraries/freertos_plus/standard/freertos_plus_posix/include \
```
Comment (yuhui-zheng) [inline doc/config/posix]: FreeRTOS_POSIX_internal.h under this directory is better not to be documented.

File: doc/config/posix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        libraries/freertos_plus/standard/freertos_plus_posix/source \
+		libraries/abstractions/posix/include/FreeRTOS_POSIX/sys \
```
Comment (yuhui-zheng) [inline doc/config/posix]: Why ```libraries/abstractions/posix/include/FreeRTOS_POSIX/sys``` only? What about ```libraries/abstractions/posix/include/FreeRTOS_POSIX```?

File: doc/config/posix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		vendors/pc/boards/windows/ports/posix \
```
Comment (yuhui-zheng) [inline doc/config/posix]: I doubt about port specific configs. It's nice to show some example, but may seem unnecessary...

File: doc/config/posix (hunk 2 above)
Comment (cobusve) [reply doc/config/posix]: Yes I agree, only document the public API's

File: doc/guidance.md [hunk 5]
```diff
@@ -14,9 +14,10 @@ This is about how to use Doxygen to maintain the API documents. There are three
 	- on mac, run "brew install graphviz"
 - goto the root directory of "amazon-freertos" (this will be referenced as $ROOT_AFR_DIR)
 	- cd $ROOT_AFR_DIR
-- run Doxygen command to generate the documents 
+- run Doxygen command to generate the documents (You might need to create tag folder manually and run the script twice)
```
Comment (cobusve) [inline doc/guidance.md]: This is good short advice but I would prefer if we explained why also, maybe just say "run the script twice to ensure all the tag files are available to complete links" or something like that?

File: libraries/freertos_plus/standard/freertos_plus_posix/source/FreeRTOS_POSIX_pthread_cond.c [hunk 6]
```diff
@@ -77,6 +77,25 @@ static void prvInitializeStaticCond( pthread_cond_internal_t * pxCond )
         taskEXIT_CRITICAL();
     }
 }
+/* @brief Check "atomically" if iLocalWaitingThreads == pxCond->iWaitingThreads and decrement.
```
Comment (cobusve) [inline libraries/freertos_plus/standard/freertos_plus_posix/source/FreeRTOS_POSIX_pthread_cond.c]: I believe it has to start with /** (two stars) for Doxygen to pick up the brief?

Comment (huguesBouvier) [reply libraries/freertos_plus/standard/freertos_plus_posix/source/FreeRTOS_POSIX_pthread_cond.c]: Fixed!

File: doc/guidance.md (hunk 5 above)
Comment (huguesBouvier) [reply doc/guidance.md]: You are right. Fixed.

File: doc/config/posix (hunk 4 above)
Comment (huguesBouvier) [reply doc/config/posix]: I agree. Removed!

File: doc/config/posix (hunk 1 above)
Comment (huguesBouvier) [reply doc/config/posix]: MQTT document everything so I want to keep it consistent.

File: doc/config/posix (hunk 2 above)
Comment (huguesBouvier) [reply doc/config/posix]: MQTT document everything so I want to keep it consistent.

File: libraries/freertos_plus/standard/freertos_plus_posix/source/FreeRTOS_POSIX_pthread_cond.c [hunk 7]
```diff
@@ -194,13 +216,21 @@ int pthread_cond_timedwait( pthread_cond_t * cond,
...
 {
+    unsigned iLocalWaitingThreads;
     int iStatus = 0;
     pthread_cond_internal_t * pxCond = ( pthread_cond_internal_t * ) ( cond );
     TickType_t xDelay = portMAX_DELAY;
 
     /* If the cond is uninitialized, perform initialization. */
     prvInitializeStaticCond( pxCond );
 
+    if( xSemaphoreTake( ( SemaphoreHandle_t ) &pxCond->xCondWaitSemaphore,
```
Comment (yuhui-zheng) [inline libraries/freertos_plus/standard/freertos_plus_posix/source/FreeRTOS_POSIX_pthread_cond.c]: Hmm is this change intended? mutex is locked when entering this function to start with...
